CPP_STYLE_GUIDE.md

```diff
@@ -192,40 +192,35 @@ class FancyContainer {
 
 ### Use `nullptr` instead of `NULL` or `0`
 
-What it says in the title.
+Further reading at [ES.47].
 
 ### Ownership and Smart Pointers
 
-"Smart" pointers are classes that act like pointers, e.g.
-by overloading the `*` and `->` operators. Some smart pointer types can be
-used to automate ownership bookkeeping, to ensure these responsibilities are
-met. `std::unique_ptr` is a smart pointer type introduced in C++11, which
-expresses exclusive ownership of a dynamically allocated object; the object
-is deleted when the `std::unique_ptr` goes out of scope. It cannot be
-copied, but can be moved to represent ownership transfer.
-`std::shared_ptr` is a smart pointer type that expresses shared ownership of a
-dynamically allocated object. `std::shared_ptr`s can be copied; ownership
-of the object is shared among all copies, and the object
-is deleted when the last `std::shared_ptr` is destroyed.
-
-Prefer to use `std::unique_ptr` to make ownership
-transfer explicit. For example:
+[R.20]: Use `std::unique_ptr` or `std::shared_ptr` to represent ownership
+[R.21]: Prefer `unique_ptr` over `shared_ptr` unless you need to share ownership
 
+Use `std::unique_ptr` to make ownership transfer explicit. For example:
 ```cpp
 std::unique_ptr<Foo> FooFactory();
 void FooConsumer(std::unique_ptr<Foo> ptr);
 ```
+Since `std::unique_ptr` has only move semantics, passing one by value trasfers
+ownership to the callee and invalidates the instance the caller hold.
 
-Never use `std::auto_ptr`. Instead, use `std::unique_ptr`.
+Avoid `std::auto_ptr`. It was deprecated in C++11 and removed from C++17.
 
 ## Others
 
 ### Type casting
 
-- Always avoid C-style casts (`(type)value`)
-- `dynamic_cast` does not work because RTTI is not enabled
-- Use `static_cast` for casting whenever it works
-- `reinterpret_cast` is okay if `static_cast` is not appropriate
+- Use `static_cast<T>` if casting is required, and its valid 
+- Use `reinterpret_cast` only when its essential
```

Comment:
typo: it is (twice)

(and maybe s/essential/necessary/)

Reply:
ack x 3